tmt/utils.py

```diff
@@ -4856,19 +4864,53 @@ def git_clone(
 
     if can_change:
         url = clonable_git_url(url)
+
+    def get_env(env: str) -> Optional[int]:
+        """ Get the value of an environment variable and convert it to be integer """
+        out = None
+        value = os.environ.get(env, None)
+        if value is not None:
+            try:
+                out = int(value)
+            except ValueError:
+                out = None
+        return out
+
+    timeout = timeout or get_env('TMT_GIT_CLONE_TIMEOUT')
+    attempts = attempts or get_env('TMT_GIT_CLONE_ATTEMPTS')
+    if attempts is None:
+        attempts = DEFAULT_GIT_CLONE_ATTEMPTS
+    interval = interval or get_env('TMT_GIT_CLONE_INTERVAL')
+    if interval is None:
+        interval = DEFAULT_GIT_CLONE_INTERVAL
+
     try:
         return common.run(
             Command(
                 'git', 'clone',
                 *depth,
                 url, destination
-                ), env=env)
+                ), env=env, timeout=timeout)
     except RunError:
         if not shallow:
             # Do not retry if shallow was not used
             raise
-        # Git server might not support shallow cloning, try again (do not modify url)
-        return git_clone(url, destination, common, env, shallow=False, can_change=False)
+        # Note that the `url` is changed even if `can_change` is `True` when the function is
+        # called at the first time, it is not necessary to changed it again when retrying.
+        # Hence, `can_change` is always set as `False` right here.
+        return retry(
```

Comment:
maybe better to add argument names? especially as you are giving the arguments on separate lines anyway